instructor/ui/instructor.js

```diff
@@ -751,8 +842,315 @@ $(function() {
       window.location = '/ab/assignments/microscopy_analyze/';
     });
   });
-  
-  $("input[id='id_objective']").attr('placeholder', '20x');
-});
 
+  $("input[id='id_objective']").attr('placeholder', '20x'); // Add default objective to the input field
 
+  $("input[id^='tech_has_']").click(function() {
+      var technique = $(this).attr('id').replace('tech_', '');
+      var body = {'technique': technique};
+      $.ajax({
+          url: "/ab/assignments/select_technique/",
+          type: "POST",
+          contentType: "application/json",
+          data: JSON.stringify(body),
+          success: function (data) {
+              console.log("Technique is chosen", data);
+              location.href = data;
+          },
+          error: function (xhr, textStatus, err){
+              console.log("Error adding Technique: " + err + " " + textStatus);
+          }
+      });
+  });
+
+  /*
+    Drag&Drop functionality for the image(s) uploading process in the micro analyze technique.
+   */
+
+  var $filesBox = $('.scb_ab_s_image_bank'),
+      $imageForm = $('.scb_ab_s_image_form'),
+      $fileInput = $imageForm.find('input[type="file"]'),
+      $label = $imageForm.find('label[for="id_file"]'),
+      $fileInput = $imageForm.find( 'input[type="file"]'),
+      $filesSelect = $(".scb_ab_s_select_box"),
+      droppedFiles = false,
+      // Function changing label during uploading process
+      showFiles = function(files, html) {
+          if (html) {
+            $label.html(html);
+          } else {
+            $label.text((files.length > 1 ? ($fileInput.attr('data-plural-caption') || '')
+                .replace('{}', files.length) : files[0].name + " is ") + "uploading ...");
+          }
+      };
+  // New event handler for the 'usual' file uploading (not drag&drop)
+  $fileInput.on('change', function(e) {
+      showFiles(e.target.files);
+      $imageForm.trigger('submit');
+  });
+  // Event handler for drag&drop uploading workflow
+  $filesBox.on('dragend dragover dragenter dragleave drop', function(e) {
+    e.preventDefault();
+    e.stopPropagation();
```

Comment:
Could just use

`return false;`

instead of:

```
e.preventDefault();
e.stopPropagation();
```

Reply:
improved